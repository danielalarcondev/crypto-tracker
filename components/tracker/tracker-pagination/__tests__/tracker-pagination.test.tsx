import * as React from 'react';
import '@testing-library/jest-dom/jest-globals';

import { fireEvent, render, screen } from '@testing-library/react';
import { expect, it, describe, beforeEach } from '@jest/globals';
import TrackerPagination from '@components/tracker/tracker-pagination/tracker-pagination';

describe('TrackerPage: ', () => {

    beforeEach(() => (
        global.ResizeObserver = jest.fn().mockImplementation(() => ({
            observe: jest.fn(),
            unobserve: jest.fn(),
            disconnect: jest.fn(),
        }))
    ));

    it('should render correctly', () => {

        const { container } = render(<TrackerPagination currentPage={1} setCurrentPage={jest.fn()} totalPages={100} />);
        expect(container).toMatchSnapshot();
    });

    it('should use properly navigation callback', async () => {

        const setCurrentPageMock = jest.fn();

        render(<TrackerPagination currentPage={1} setCurrentPage={setCurrentPageMock} totalPages={100} />);

        const button2 = await screen.findByText('2');
		
        fireEvent.click(button2);
        expect(setCurrentPageMock).toHaveBeenCalled();
    });

    it('should render proper amount of pages', async () => {

        const TWO = 2;
        const setCurrentPageMock = jest.fn();
        const totalPages = 50;

        render(<TrackerPagination currentPage={1} setCurrentPage={setCurrentPageMock} totalPages={totalPages} />);

        const button50 = await screen.findByText(totalPages);
        const ul = button50.parentElement?.parentElement;
        const penultimateUlChild = ul?.children[ul?.children.length - TWO];

        expect(penultimateUlChild?.firstChild).toHaveTextContent(totalPages.toString());
    });
});
