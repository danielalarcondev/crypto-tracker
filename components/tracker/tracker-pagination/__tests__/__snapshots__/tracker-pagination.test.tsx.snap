// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackerPage:  should render correctly 1`] = `
<div>
  <div
    class="my-6 mx-auto w-56 sm:w-[32rem] min-h-10"
    data-testid="tracker-pagination"
  >
    <ul
      class="inline-flex w-full justify-center"
      style=""
    >
      <li
        class="w-8 h-10 rounded-full mr-2 shrink-0 pointer hover:bg-primary-500/10 disabled border-none text-3xl navigation-span-pointer hover:scale-110"
      >
        <span
          aria-label="Previous"
          class="w-full h-full flex justify-center content-center flex-wrap shrink-0"
        >
          <span
            aria-hidden="true"
          >
            «
          </span>
        </span>
      </li>
      <li
        aria-current="page"
        class="w-8 h-10 rounded-full mr-2 shrink-0 pointer hover:bg-primary-500/10 bg-primary-500 hover:!bg-primary-500 text-white"
      >
        <a
          class="w-full h-full flex justify-center content-center flex-wrap shrink-0"
          href="#"
        >
          1
        </a>
      </li>
      <li
        class="w-8 h-10 rounded-full mr-2 shrink-0 pointer hover:bg-primary-500/10"
      >
        <a
          class="w-full h-full flex justify-center content-center flex-wrap shrink-0"
          href="#"
        >
          2
        </a>
      </li>
      <li
        aria-hidden="true"
        class="w-8 h-10 rounded-full mr-2 shrink-0 pointer hover:bg-primary-500/10 disabled"
      >
        <span
          class="w-full h-full flex justify-center content-center flex-wrap shrink-0"
        >
          …
        </span>
      </li>
      <li
        class="w-8 h-10 rounded-full mr-2 shrink-0 pointer hover:bg-primary-500/10"
      >
        <a
          class="w-full h-full flex justify-center content-center flex-wrap shrink-0"
          href="#"
        >
          100
        </a>
      </li>
      <li
        class="w-8 h-10 rounded-full mr-2 shrink-0 pointer hover:bg-primary-500/10 border-none text-3xl navigation-span-pointer hover:scale-110"
      >
        <a
          aria-label="Next"
          class="w-full h-full flex justify-center content-center flex-wrap shrink-0"
          href="#"
        >
          <span
            aria-hidden="true"
          >
            »
          </span>
        </a>
      </li>
    </ul>
  </div>
</div>
`;
